src/Surfnet/Stepup/Identity/EventSourcing/IdentityRepository.php

```diff
@@ -18,32 +18,76 @@
 
 namespace Surfnet\Stepup\Identity\EventSourcing;
 
+use Broadway\Domain\DomainMessage;
 use Broadway\EventHandling\EventBus as EventBusInterface;
 use Broadway\EventSourcing\AggregateFactory\AggregateFactory;
 use Broadway\EventSourcing\EventSourcingRepository;
 use Broadway\EventSourcing\EventStreamDecorator;
 use Broadway\EventStore\EventStore as EventStoreInterface;
+use Broadway\EventStore\EventStreamNotFoundException;
+use Broadway\Repository\AggregateNotFoundException;
+use Broadway\Serializer\Serializable;
+use Psr\Log\LoggerInterface;
+use Surfnet\Stepup\Identity\Identity;
+use Surfnet\Stepup\Identity\Value\IdentityId;
+use Surfnet\StepupMiddleware\CommandHandlingBundle\SensitiveData\Forgettable;
 
 class IdentityRepository extends EventSourcingRepository
 {
+    protected $events;
+
+    protected $logger;
+
     /**
-     * @param EventStoreInterface             $eventStore
-     * @param EventBusInterface               $eventBus
-     * @param AggregateFactory       $aggregateFactory
      * @param EventStreamDecorator[] $eventStreamDecorators
      */
     public function __construct(
         EventStoreInterface $eventStore,
         EventBusInterface $eventBus,
         AggregateFactory $aggregateFactory,
+        LoggerInterface $logger,
         array $eventStreamDecorators = []
     ) {
+        $this->events = $eventStore;
+        $this->logger = $logger;
         parent::__construct(
             $eventStore,
             $eventBus,
-            \Surfnet\Stepup\Identity\Identity::class,
+            Identity::class,
             $aggregateFactory,
             $eventStreamDecorators
         );
     }
+
+    public function obtainInformation(IdentityId $id): array
+    {
+        try {
+            $domainEventStream = $this->events->load($id);
+            $data = [];
+            /** @var DomainMessage $domainMessage */
+            foreach ($domainEventStream as $domainMessage) {
+                $event = $domainMessage->getPayload();
+                if (!$event instanceof Serializable) {
```

Comment:
Should this also be checked also for the sensitive data?

Reply:
In parallel, @VadimSchmitz is working on a means to retrieve the user data directly from the Events. A next PR will integrate those changes here. Removing this 'pseudo' code that is currently in place.